blueprints/federations.redesign.md

```diff
@@ -0,0 +1,546 @@
+Services" and (conditionally) "Delivery Service Requests" titled "Federations".
+This menu item directs users to a page with a standard table for Federations
+just like the tables for every other API object. Within a single Federation's
+page is a form for editing that Federation, a sample of which is shown below.
+
+![Mock UI for Federations form](img/federations.form.png)
+
+Note that Federation Resolvers are manipulated here directly, rather than
+created separately through a different page and then linked using a popup.
+
+To accommodate re-use of Federation Resolvers, the "More" menu for a Federation
+will have two options: "Copy Resolvers From..." and "Clone Resolvers To...".
+These buttons will open dialog boxes that will prompt the user to choose a
+Federation from which to copy Resolvers onto the Federation being edited, and
+to choose one or more Federations to which to copy the Federation being edited's
+Resolvers, respectively. Sample Dialogs for each are shown below.
+
+![Mock UI for Federation Copy Resolvers From dialog](img/federations.copy.from.dialog.png)
+
+![Mock UI for Federation Clone Resolvers To dialog](img/federations.clone.to.dialog.png)
+
+(Note that these feature a "Replace Existing Resolvers?" checkbox, which has
+since been reconsidered and should not actually be included in the UI - instead
+the resolvers should *always* be overwritten)
+
+If desired, the markup used for these samples may be found in
+[Appendix A](#appendix-a).
+
+### Traffic Router Impact
+Traffic Router's Federations polling will need to be stripped out, and instead
+Federation information should be parsed from the CDN Snapshot.
+
+## Documentation Impact
+The documentation for affected endpoints will need to be updated, and a section
+for the new property of Delivery Services will need to be added to the Delivery
+Service overview section. Also, there is a "Quick How-To" guide for setting up
+Federations that currently still references the Traffic Ops UI that hasn't
+existed for years; that needs to be updated either way, but since the changes to
+Federations described in this blueprint include Traffic Portal changes, it would
+be a good time to also update the guide. Assuming it's quickly approved.
+
+## Testing Impact
+The tests for Federation-related endpoints will need to be overhauled to remove
+tests for removed endpoints, and to reflect changes to the updated endpoints.
+
+## Performance Impact
+The changes herein proposed would make Snapshots larger, but would eliminate an
+entire polling thread from Traffic Router. The performance impact to Traffic
+Router is difficult to know without measuring (and I haven't measured, not only
+because that would require actually making the changes to compare), but I would
+guess that Traffic Router will use less resources overall with this tradeoff.
+
+Costs/gains to Traffic Ops and/or Traffic Portal are negligible.
+
+## Security Impact
+None.
+
+## Upgrade Impact
+None.
```

Comment:
What I'm saying is that by not breaking/removing this route that TR uses, we don't have to add in a bunch of unnecessary code to handle that backwards-compatibility, all for the sake of reducing the number of routes in the federations API by 1. If there were stronger reasons for this API to be removed, then of course we'd have to make it backwards-compatible. I just don't think simplifying the federations API by removing this route justifies making TR more complex as a result.

Reply:
No, for the sake of these changes I think leaving `/federations/all` as a full mapping of xmlids to resolver groups is probably the way to go, I'm just concerned that TR as-is can't handle backwards-compatibility, because all components (besides TP) are supposed to, to prevent enforcing an upgrade order. Just a general concern, not related to these changes.